src/dune_rules/scope.ml

```diff
@@ -265,14 +268,16 @@ module DB = struct
                 let entry =
                   match stanza.package with
                   | None -> Coq_lib.DB.Theory dir
-                  | Some _ -> Redirect public_theories
+                  | Some _ -> Redirect (Lazy.force public_theories)
```

Comment:
Might as well go the whole way and make coq_db_by_project_dir map lazily constructed.

Reply:
How's that gonna work for merging the map with `db_by_project_dir` ? We still need access to the keys.

Reply:
Instead of merging, you can just look it up after forcing the map:

```ocaml
let coq_db = lazy (Path.Source.Map.find_exn (Lazy.force coq_map) dir)) in ...
```

Reply:
Ok, that's a bit deeper change, let's see how it looks like.

Reply:
Done

Reply:
Turned out great in the end IMHO.

Reply:
Ok I found the cause, the problem is that now library DB creation is too lazy, and the install rules run first emitting the more confusing error:
```  
$ dune build B
 |  Error: Cannot have more than one boot theory in scope:
 |  - Coq at Coq/dune:1
 |  - Coq at B/Coq/dune:2
+|  -> required by alias B/default
+|  Error: Multiple rules generated for
+|  _build/install/default/lib/coq/theories/Init/Prelude.v:
+|  - Coq/dune:1
+|  - B/Coq/dune:2
+|  -> required by _build/default/B/Foo.install
+|  -> required by alias B/all
+|  -> required by alias B/default
```
I'm tempted to leave this like they are in this PR, even with the worse error message. WDTY @rgrinberg ?

Reply:
Fine by me. You can also easily fix bring back the old behavior if you force the coq scope when setting up install rules for a theory.

Reply:
Indeed the latter option seems the best for me, thanks.